packages/legend-query/src/components/QueryBuilderPropertyExpressionEditor.tsx

```diff
@@ -64,14 +74,27 @@ const DerivedPropertyParameterValueEditor = observer(
       },
       [derivedPropertyExpressionState, idx],
     );
+    const isDraggedTypeCompatibleWithTarget = (
```

Comment:
I think we can inline this instead of creating this function.

Reply:
See the previous comment from Mauricio: `this code is kind of hard to read. Could we maybe abstract this out and name it exactly what we are doing. We checking compatiblity of types right?`
Let me know if we need to inline this

Reply:
yeah, @MauricioUyaguari suggested one way of making the code more readable, I would argue that abstracting it with this minute piece of logic make code equally hard to read. Instead, what we should do here is to inline and write comment for that condition.